app/controllers/api/v1/authentication_controller.rb

```diff
@@ -190,6 +186,31 @@ def auth_params
       params.require(:user_auth).permit(attributes)
     end
 
+    def valid_user
+      valid_mobile? && User.find_by_mobile(@mobile)
+    end
+
+    def valid_mobile?
+      User::HongKongMobileRegExp === @mobile
+    end
+
+    def render_invalid_mobile
+      attr = I18n.t('activerecord.attributes.user.mobile')
+      reason = @mobile.blank? ? 'blank' : (valid_mobile? ? 'not_registerd' : 'invalid')
```

Comment:
Line is too long. [87/80]<br>Ternary operators must not be nested. Prefer `if`/`else` constructs instead.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.